sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/Earcut.cs

```diff
@@ -667,13 +676,21 @@ public static Data Flatten(List<List<float[]>> data)
 		
 		public static Data Flatten(List<List<Vector3>> data)
 		{
-			var dim = 2;
-			var result = new Data() { Dim = dim };
+			var dataCount = data.Count;
+			var dim = 0;
+			for (int i = 0; i < dataCount; i++)
+			{
+				dim += data[i].Count;
+			}
+
+			var result = new Data() { Dim = 2 };
```

Comment:
Why `Dim = 2` ? If we are setting `Dim` to a hardcoded value, what is the use of the for-loop above to compute `dim` ?

Reply:
I'm not sure what's up with this. I'm on another branch at the moment but it isn't like this. Maybe mistakenly broken it like that but I have no idea how that could be. 
Anyway dim is always 2 for our case as our raw data is two dimensional anyway so I hardcoded that. I'll switch and check this again as well.

Reply:
ah now I get it, it's not a mistake, it really is optimization :) 
idea is to calculate the vertex count *2 so set the "result.Vertices" size buuut it's using "dim" field to not initialize yet another int field. 
"dim" field is use to set dimension count in first float overload (original one, not used).
"dim" field isn't even used in vector3 overload.
I'll just rename the second one. and maybe delete that first function all together?